src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,126 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private transient Object[] elementData;
+    private int size;
+
+    public ArrayList() {
+        elementData = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == elementData.length) {
+            growElementData();
+        }
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        rangeCheck(index, true);
+        if (size == elementData.length) {
+            growElementData();
+        }
+        System.arraycopy(elementData, index,
+                elementData, index + 1,
+                size - index);
+        elementData[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        rangeCheck(index, false);
+        return (T) elementData[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        rangeCheck(index, false);
+        elementData[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        rangeCheck(index, false);
+        T oldValue = (T) elementData[index];
+        removeElement(elementData, index);
+        return oldValue;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        final int size = this.size;
+        int i = 0;
+        found:
+        {
+            if (element == null) {
+                for (; i < size; i++) {
+                    if (elementData[i] == null) {
+                        break found;
+                    }
+                }
+            } else {
+                for (; i < size; i++) {
+                    if (element.equals(elementData[i])) {
+                        break found;
+                    }
+                }
+            }
+            throw new NoSuchElementException();
+        }
+        T oldValue = (T) elementData[i];
+        removeElement(elementData, i);
+        return oldValue;
```

Comment:
Please, try to simplify this method. It's difficult to read and understand it.
Try to avoid nested "if"